Buildings/Fluid/Storage/Plant/ChillerAndTank.mo

```diff
@@ -0,0 +1,418 @@
+within Buildings.Fluid.Storage.Plant;
+model ChillerAndTank
+  "(Draft) Model of a plant with a chiller and a tank where the tank can potentially be charged remotely"
+  extends Buildings.Fluid.Interfaces.PartialFourPortInterface(
+    final m1_flow_nominal = m2_flow_nominal,
+    final m2_flow_nominal = mChi_flow_nominal + mTan_flow_nominal);
+
+  parameter Boolean allowRemoteCharging = true
+    "Turns the plant to a prosumer";
+
+  parameter Modelica.Units.SI.MassFlowRate mChi_flow_nominal=1
+    "Nominal mass flow rate for the chiller branch";
+  parameter Modelica.Units.SI.MassFlowRate mTan_flow_nominal=1
+    "Nominal mass flow rate for the tank branch";
+  parameter Modelica.Units.SI.PressureDifference dp_nominal=
+    p_CHWS_nominal-p_CHWR_nominal
+    "Nominal pressure difference";
+  parameter Modelica.Units.SI.AbsolutePressure p_CHWS_nominal=800000
+    "Nominal pressure of the CHW supply line";
+  parameter Modelica.Units.SI.AbsolutePressure p_CHWR_nominal=300000
+    "Nominal pressure of the CHW return line";
+  parameter Modelica.Units.SI.Temperature T_CHWS_nominal=7+273.15
+    "Nominal temperature of CHW supply";
+  parameter Modelica.Units.SI.Temperature T_CHWR_nominal=12+273.15
+    "Nominal temperature of CHW return";
+
+  replaceable parameter Buildings.Fluid.Chillers.Data.ElectricEIR.Generic perChi(
+    QEva_flow_nominal=-1E6,
+    COP_nominal=3,
+    PLRMax=1,
+    PLRMinUnl=0.3,
+    PLRMin=0.3,
+    etaMotor=1,
+    mEva_flow_nominal=mChi_flow_nominal,
+    mCon_flow_nominal=mChi_flow_nominal*1.2,
+    TEvaLvg_nominal=280.15,
+    capFunT={1,0,0,0,0,0},
+    EIRFunT={1,0,0,0,0,0},
+    EIRFunPLR={1,0,0},
+    TEvaLvgMin=276.15,
+    TEvaLvgMax=288.15,
+    TConEnt_nominal=310.15,
+    TConEntMin=303.15,
+    TConEntMax=333.15)
+    "Chiller performance data"
+    annotation (choicesAllMatching=true,
+      Placement(transformation(extent={{-100,82},{-80,102}})));
+
+  Buildings.Fluid.FixedResistances.PressureDrop preDro2(
+    redeclare package Medium = Medium2,
+    final allowFlowReversal=true,
+    final dp_nominal=dp_nominal/10,
+    final m_flow_nominal=mTan_flow_nominal) "Flow resistance on tank branch"
+    annotation (Placement(transformation(extent={{10,-10},{-10,10}},
+        rotation=-90,
+        origin={-80,-80})));
+  Modelica.Blocks.Interfaces.RealInput set_mPum1_flow
+    "Primary pump mass flow rate setpoint" annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=-90,
+        origin={40,130}),  iconTransformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={-110,100})));
+
+  Buildings.Fluid.Movers.FlowControlled_m_flow pum1(
+    redeclare package Medium = Medium2,
+    per(pressure(dp=dp_nominal*{2,1.2,0}, V_flow=mChi_flow_nominal/1.2*{0,1.2,2})),
```

Comment:
@bravache 
Can you please explain what this means?

Reply:
Sure. When I say "expose", I mean have that parameter be declared in the model and then propagated through to that pump:
```modelica
replaceable parameter Buildings.Fluid.Movers.Data.Generic perPum(
    pressure(
        dp=dp_nominal*{2,1.2,0}, 
        V_flow=mChi_flow_nominal/1.2*{0,1.2,2})
) "Pump performance";

[...]

  Buildings.Fluid.Movers.FlowControlled_m_flow pum1(
    redeclare package Medium = Medium2,
    final per = perPum)
```

This way, if a user wants to specify its own pump curve, they can do so directly by parameterizing `ChillerAndTank` like so:

```modelica
ChillerAndTank(perPum = myPerPum)
```

rather than having to go one layer down:

```modelica
ChillerAndTank(pum(per = myPerPum))
```

It's not more efficient, but typically parameters that are relevant to the system behavior should be exposed to user, so that they don't need to know how the model is built to instantiate their value.